### PR TITLE
Some more docs and testing

### PR DESCRIPTION
**Pull request by grigi**
Refactoring:
* [x] Refactored MANIFEST and setup.py to include only required code from sdist. (excludes tests .pyc files)
* [x] Refactored the unittest classes.
* [x] Started applying some type annotations
* [x] `__slots__` where it was easy to do. (turns class instances into tuples versus dicts, mostly massive memory savings)

Tests:
* [x] Wrote tests for unittest classes.
* [x] Wrote tests for `tortoise.backends.base.db_url`.
* [x] Wrote tests for QuerySet methods.

Docs:
* [x] Expanded docs for Exceptions
* [x] Expanded docs for QuerySet
* [x] Split up Models and Fields in documentation.
* [x] Split out init() documentation
* [x] Added roadmap and styleguide

I feel that this PR is getting too large, as it is changing a lot of things already.

### CODE REVIEW

**Comment by grigi** [comment]
@Zeliboba5 I feel that this PR is getting too large, as it is changing a lot of things already. Please review.

File: tortoise/__init__.py [hunk 1]
```diff
@@ -70,15 +69,15 @@ def _init_relations(cls):
                         raise ConfigurationError(
                             'backward relation "{}" duplicates in model {}'.format(
                                 backward_relation_name, related_model_name))
-                    relation = fields.BackwardFKRelation(model, '{}_id'.format(field))
-                    setattr(related_model, backward_relation_name, relation)
+                    fkrelation = fields.BackwardFKRelation(model, '{}_id'.format(field))
+                    setattr(related_model, backward_relation_name, fkrelation)
                     related_model._meta.filters.update(
-                        get_backward_fk_filters(backward_relation_name, relation)
+                        get_backward_fk_filters(backward_relation_name, fkrelation)
```
Comment (Zeliboba5) [inline tortoise/__init__.py]: How about doing it like `fk_relation` and `m2m_relation`? I think it always easier to read it like that, than read concatted words

Comment (grigi) [reply tortoise/__init__.py]: Yes, this is raised more than once. It should go into a project style guide.
That was me changing the variable names to make mypy less confused. I'll fix it now.

**Comment by Zeliboba5** [comment]
👍

**Pull request merged**